assets/js/dashboard.js

```diff
@@ -79,49 +77,78 @@ function renderDashboardResults(searchQuery) {
         .then(data => {
             console.log(data); // SEE: data format.
-            locationDesc.innerHTML = data.items[0].htmlSnippet.concat(" <a href=", data.items[0].formattedUrl, ">Read More</a><br>Result retrieved from ", data.items[0].displayLink)/* City info from API */;
+            locationDesc.innerHTML = data.items[0].htmlSnippet.concat(" <a href=", data.items[0].formattedUrl, ">Read More</a><br>Result retrieved from ", data.items[0].displayLink);/* City info from API */;
         })
         .catch(error => {
-            console.log(error);
+            console.log("Error:", error); // prints the error in the console
         });
 
-    // API fetch request | "https://www.googleapis.com/customsearch/v1?key={API key}&cx={PSE ID}&q={city name}&searchType=image&num=1"
-    // This is for a Google Images SafeSearch.
-    let fetchURLGoogleImage = googleAPIURL.concat("key=", googleAPIKeyAN, "&cx=", googlePSEID, "&q=", searchQuery, "&searchType=image&num=1")
-    fetch(fetchURLGoogleImage)
-        .then(response => {
-            if (!response.ok) {
-                throw new Error("API response was not okay. (Google images query)");
-            }
-            return response.json();
-        })
+    fetchWithRetry(googleAPIURL.concat("key=", googleAPIKeyAN, "&cx=", googlePSEID, "&q=", searchQuery, "&searchType=image&num=1"))
         .then(data => {
             console.log(data); // SEE: data format.
-            locationImage.src = data.items[0].link
+            locationImage.src = data.items[0].link;
         })
         .catch(error => {
-            console.log(error);
+            console.log("Error:", error); // prints the error in the console
         });
+}
+
+// Function to extract 5-day forecast data
+function get5DayForecast(dataList) {
+    let daysData = [];
+    let currentDate = "";
+
+    dataList.forEach(item => {
+        let date = item.dt_txt.split(" ")[0];
+        if (date !== currentDate) {
+            currentDate = date;
+            daysData.push({
+                date: new Date(date).toLocaleDateString(),
+                temp: item.main.temp,
+                weather: item.weather[0].description
+            });
+        }
+    });
+
+    return daysData.slice(0, 5); // Ensure only 5 days are returned
+}
+
+// Function to display 5-day forecast data
+function display5DayForecast(daysData) {
+    let forecastContainer = document.getElementById("forecast-container");
+    forecastContainer.innerHTML = "";
+
+    daysData.forEach(dayData => {
+        let forecastDiv = document.createElement("div");
+        forecastDiv.className = "forecast-day";
+        forecastDiv.innerHTML = `
+            <h3>${dayData.date}</h3>
+            <p>Temp: ${temperatureToCelcius(dayData.temp)}</p>
+            <p>Weather: ${dayData.weather}</p>
+        `;
+        forecastContainer.appendChild(forecastDiv);
+    });
+}
+// Function to display current wind speed
+function displayWindSpeed(speed, elementId) {
```

Comment:
Great work with this function, however, it is not being called anywhere at the moment, thus, is not being executed. Feel free to implement code to address this.